### PR TITLE
Ensure the slice is an array

### LINKED ISSUE #95: Error when using xarray.DataArray and gsw.geo_strf_dyn_height with numpy < 1.23
When using xarray.DataArray as inputs for gsw.geo_strf_dyn_height, the result is only NaNs.
Do you have any idea why this is so?

# Example
```python
import gsw
a = [0,1,2]
print(gsw.geo_strf_dyn_height(a, a, a))
```
Outputs
```
array([ 0.        , -0.27061522, -0.53269375])
```

Using xarray
```python
import gsw
import xarray as xr
da = xr.DataArray([0,1,2])
print(gsw.geo_strf_dyn_height(da, da, da))
```
Outputs
```
array([nan, nan, nan])
```

With numpy 1.22.2 I also get a FutureWarning:
```
/home/romain/.cache/pypoetry/virtualenvs/gsw-xarray-NsrEXKiZ-py3.8/lib/python3.8/site-packages/gsw/geostrophy.py:83: FutureWarning: Using a non-tuple sequence for multidimensional indexing is deprecated; use `arr[tuple(seq)]` instead of `arr[seq]`. In the future this will be interpreted as an array index, `arr[np.array(seq)]`, which will result either in an error or a different result.
  pgood = p[ind][igood]
```

### PR DESCRIPTION
**Pull request by ocefpaf**
@DocOtak I'm not sure if we can do something fancier here than just casting to an array. Do you have any suggestions?

Closes #95

### CODE REVIEW

**Comment by DocOtak** [comment]
It kinda sounds like numpy is unhappy with whatever the 'indexer' function is returning?

File: gsw/geostrophy.py [hunk 1]
```diff
@@ -78,7 +78,7 @@ def geo_strf_dyn_height(SA, CT, p, p_ref=0, axis=0, max_dp=1.0,
     except AttributeError:
         order = 'C'  # e.g., xarray DataArray doesn't have flags
     for ind in indexer(SA.shape, axis, order=order):
-        igood = goodmask[ind]
+        igood = np.array(goodmask[ind])
```
Comment (efiring) [inline gsw/geostrophy.py:81]: I think that using `np.asarray` would be more efficient, avoiding the copy which is unnecessary when the original function arguments are numpy arrays.  I would also add a comment that this is needed to support xarray inputs for numpy < 1.23.

**Review by efiring — CHANGES_REQUESTED** [review]

File: gsw/geostrophy.py (hunk 1 above)
Comment (ocefpaf) [reply gsw/geostrophy.py]: Done.

**Review by efiring — COMMENTED** [review]
The reason I suggested `asarray` rather than `asanyarray` is that the point is to end up with a plain ndarray as the boolean indexing object.  It's unlikely to make any difference in practice, since masked arrays will never end up here and DataArrays (and Pandas structures, for that matter) are not ndarray subclasses, at least in recent versions; but I think that `asarray` still has the advantage of clarifying the intent.

**Comment by ocefpaf** [comment]
> The reason I suggested `asarray` rather than `asanyarray` is that the point is to end up with a plain ndarray as the boolean indexing object.

VScode completed to asany and I did not even noticed! I literally copied-and-pasted from your comment. Fixing it in a moment...

**Review by efiring — APPROVED** [review]

**Pull request merged**